diff --git a/src/focustracker.js b/src/focustracker.js
index 9870efa8b2c2273a96ec4f57c22e5ac974aaffe4..c0f04eaed3fc39a4fce121c468dd3f414e0a7f2d 100644
--- a/src/focustracker.js
+++ b/src/focustracker.js
@@ -219,7 +219,16 @@ export class FocusTracker extends /* #__PURE__ */ DomEmitterMixin(/* #__PURE__ *
      * a focus tracker state would experience UI flashes and glitches as the user focus travels across the UI.
      */
     _blur() {
-        const isAnyElementFocused = this.elements.find(element => element.contains(document.activeElement));
+        /**
+         * Neos patch, we cannot access "document" as this will always be the host frame.
+         * If ckeditor is used in the guest iframe we need to use their document.
+         * Technically relying on the first "ownerDocument" of the possilby multiple elements is not correct as they _could_ belong to different documents but in practice they dont and should not.
+         */
+        const currentDocument = this.elements.at(0)?.ownerDocument;
+        if (!currentDocument) {
+            return;
+        }
+        const isAnyElementFocused = this.elements.find(element => element.contains(currentDocument.activeElement));
         // Avoid blurs originating from external FTs when the focus still remains in one of the #elements.
         if (isAnyElementFocused) {
             return;
@@ -263,9 +272,15 @@ function isExternalViewSubtreeFocused(subTreeRoot, view) {
     return !!view.focusTracker.externalViews.find(view => isFocusedView(subTreeRoot, view));
 }
 function isFocusedView(subTreeRoot, view) {
+    /**
+     * Neos patch, we cannot access "document" as this will always be the host frame.
+     * If ckeditor is used in the guest iframe we need to use their document.
+     * -> this patch is not as important as the blur() patch above, we could not find any side effects if this is not enabled, but better have it right.
+     */
+    const currentDocument = subTreeRoot.ownerDocument;
     // Note: You cannot depend on externalView.focusTracker.focusedElement because blurs are asynchronous and the value may
     // be outdated when moving focus between two elements. Using document.activeElement instead.
-    return !!view.element && view.element.contains(document.activeElement) && subTreeRoot.contains(view.element);
+    return !!view.element && view.element.contains(currentDocument.activeElement) && subTreeRoot.contains(view.element);
 }
 // @if CK_DEBUG_FOCUSTRACKER // declare global {
 // @if CK_DEBUG_FOCUSTRACKER // 	interface Window {
