diff --git a/src/tooltipmanager.js b/src/tooltipmanager.js
index 0c4e08c64df9a708466519a2384c550f04c98b0b..751e4be3330721df2e3e7988b7896e58c44abd63 100644
--- a/src/tooltipmanager.js
+++ b/src/tooltipmanager.js
@@ -159,12 +159,21 @@ export class TooltipManager extends /* #__PURE__ */ DomEmitterMixin() {
         });
         this._pinTooltipDebounced = debounce(this._pinTooltip, 600);
         this._unpinTooltipDebounced = debounce(this._unpinTooltip, 400);
-        this.listenTo(global.document, 'keydown', this._onKeyDown.bind(this), { useCapture: true });
-        this.listenTo(global.document, 'mouseenter', this._onEnterOrFocus.bind(this), { useCapture: true });
-        this.listenTo(global.document, 'mouseleave', this._onLeaveOrBlur.bind(this), { useCapture: true });
-        this.listenTo(global.document, 'focus', this._onEnterOrFocus.bind(this), { useCapture: true });
-        this.listenTo(global.document, 'blur', this._onLeaveOrBlur.bind(this), { useCapture: true });
-        this.listenTo(global.document, 'scroll', this._onScroll.bind(this), { useCapture: true });
+        /**
+         * Neos patch, we need to track the mouse position IN the iframe where the CKEditor is mounted as the document itself is the host frame.
+         * This patch fully opts in to only support the iframe. CKEditor mounted in the host frame - for example the RTE Editor will not show tooltips.
+         * That is the bargain. To fix both cases we need to actually boot and load CKEditor in the iframe by using code sharing.
+         */
+        const neosIframe = document.querySelector('iframe[name="neos-content-main"]')?.contentDocument;
+        if (!neosIframe) {
+            console.error('CKEditor Neos patch could not be applied missing iframe.');
+        }
+        this.listenTo(neosIframe, 'keydown', this._onKeyDown.bind(this), { useCapture: true });
+        this.listenTo(neosIframe, 'mouseenter', this._onEnterOrFocus.bind(this), { useCapture: true });
+        this.listenTo(neosIframe, 'mouseleave', this._onLeaveOrBlur.bind(this), { useCapture: true });
+        this.listenTo(neosIframe, 'focus', this._onEnterOrFocus.bind(this), { useCapture: true });
+        this.listenTo(neosIframe, 'blur', this._onLeaveOrBlur.bind(this), { useCapture: true });
+        this.listenTo(neosIframe, 'scroll', this._onScroll.bind(this), { useCapture: true });
         // Because this class is a singleton, its only instance is shared across all editors and connects them through the reference.
         // This causes issues with the ContextWatchdog. When an error is thrown in one editor, the watchdog traverses the references
         // and (because of shared tooltip manager) figures that the error affects all editors and restarts them all.
